features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/components/Shared/AppsAndKeys/ViewKeys.jsx

```diff
@@ -136,12 +136,33 @@ class ViewKeys extends React.Component {
             });
     }
 
+    /**
+     *
+     *
+     * @param {*} prevProps previous props
+     * @memberof ViewKeys
+     */
+    componentDidUpdate(prevProps) {
+        const { isKeyJWT } = this.props;
+        if (isKeyJWT && !prevProps.isKeyJWT) {
+            // eslint-disable-next-line react/no-did-update-set-state
+            this.setState({ isKeyJWT: true });
```

Comment:
Why we need to do this here ?

I think we could have done this in `render()` method, We get the same prop in `render()` and could have done them if logic check there?
And please add an explanation if adding 

```javascript
// eslint-disable-next-line react/no-did-update-set-state
```
ignore rules 😃

Reply:
Ran in to a small issue when setting in the render method. we can add it to the render method like this

`        if (jwtGenerated && !isKeyJWT) {
            this.setState({ isKeyJWT: true });
        }
`
where jwtGenerated is the what is passed from props and isKeyJWT is taken from state. But to close the dialog box i need set isKeyJWT false however if we set isKeyJWT to false it will always be set to true in the render method.
we can avoid that in componentDidUpdate with isKeyJWT && !prevProps.isKeyJWT consdition.

reason i ignored the rule is that checked with react docs and according to them its ok to setstate as long as we are checking a condition [1] 

[1] - https://reactjs.org/docs/react-component.html#componentdidupdate

Reply:
added a comment to method explaining the reason  https://github.com/wso2/carbon-apimgt/pull/6787/commits/b09339ab96cfdcc35f13e351833f406efad26ead

Reply:
Moving to render method in the sense , Not moving the logic with `this.setState`. Calling setState in `render()` method is prohibited as it could lead to infinite render loops. I think this is the case for `componentDidUpdate` method as well, But it's safe as long as we handle the infinite loop conditions there.

I think setting the state here is redundant? Do we need to update the state in every prop change ? If so , our state is not local to that component and has to be derived from props. If so [that is an anti-pattern](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html)[1], and we should avoid such scenarios.

[1] https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html

Reply:
created an issue based on discussion https://github.com/wso2/product-apim/issues/5293